### ISSUE TITLE
Can't start this example

### ISSUE THREAD

**Issue by kireerik** [callemall/material-ui-webpack-example#9]
I am getting the following errors after using the `npm start` command under Windows 10 Pro x64 with Node.js 7.9.0 and npm 4.2.0:

```powershell
ERROR in ./~/react-tap-event-plugin/src/injectTapEventPlugin.js
Module not found: Error: Cannot resolve module 'react/lib/EventPluginHub' in C:\Data\Program\Node\test\node_modules\react-tap-event-plugin\src
 @ ./~/react-tap-event-plugin/src/injectTapEventPlugin.js 23:2-37

ERROR in ./~/react-tap-event-plugin/src/TapEventPlugin.js
Module not found: Error: Cannot resolve module 'react/lib/EventConstants' in C:\Data\Program\Node\test\node_modules\react-tap-event-plugin\src
 @ ./~/react-tap-event-plugin/src/TapEventPlugin.js 22:21-56

ERROR in ./~/react-tap-event-plugin/src/TapEventPlugin.js
Module not found: Error: Cannot resolve module 'react/lib/EventPluginUtils' in C:\Data\Program\Node\test\node_modules\react-tap-event-plugin\src
 @ ./~/react-tap-event-plugin/src/TapEventPlugin.js 23:23-60

ERROR in ./~/react-tap-event-plugin/src/TapEventPlugin.js
Module not found: Error: Cannot resolve module 'react/lib/EventPropagators' in C:\Data\Program\Node\test\node_modules\react-tap-event-plugin\src
 @ ./~/react-tap-event-plugin/src/TapEventPlugin.js 24:23-60

ERROR in ./~/react-tap-event-plugin/src/TapEventPlugin.js
Module not found: Error: Cannot resolve module 'react/lib/SyntheticUIEvent' in C:\Data\Program\Node\test\node_modules\react-tap-event-plugin\src
 @ ./~/react-tap-event-plugin/src/TapEventPlugin.js 25:23-60

ERROR in ./~/react-tap-event-plugin/src/TapEventPlugin.js
Module not found: Error: Cannot resolve module 'react/lib/ViewportMetrics' in C:\Data\Program\Node\test\node_modules\react-tap-event-plugin\src
 @ ./~/react-tap-event-plugin/src/TapEventPlugin.js 27:22-58
webpack: Failed to compile.
```

I also get the following warning messages after using the `npm install` command:
```powershell
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.0.0 (node_modules\chokidar\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.1.1: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
```

However I can successfully start the [documentations](https://github.com/callemall/material-ui/tree/master/docs) including the [next branch](https://github.com/callemall/material-ui/tree/next/docs).

**Comment by oliviertassinari** [comment]
I have updated the dependencies https://github.com/callemall/material-ui-webpack-example/commit/9386d1d587345f1d7bebb7c5106514e2ca8823a6, I hope it helps.

**Comment by kireerik** [comment]
Hi @oliviertassinari,

Thank you so much!

It is working now.

I hope you don't mind if I ask some questions not closely related to this resolved issue:
- Should the `npm build` command work on a local environment?
- Do you have an example project similar to this one which supports [server side rendering](http://www.material-ui.com/#/get-started/server-rendering) and maybe HTML and CSS minification too?
- Which version of Material-UI is the newest [0.17.4 ](http://www.material-ui.com/#/) or [1.0.0-alpha.11](https://material-ui-1dab0.firebaseapp.com/)?
   - I know this may sounds a silly question, but with [1.0.0-alpha.11](https://material-ui-1dab0.firebaseapp.com/) the [Checkboxes](https://material-ui-1dab0.firebaseapp.com/component-demos/selection-controls#checkboxes)' animations become less impressive. I really like [their animation in 0.17.4](http://www.material-ui.com/#/components/checkbox).
      - Also the left side bar menu (called Drawers) become not swipeable on touch enabled devices with 1.0.0-alpha.11.
      - Plus I think [0.17.4's Documentation](http://www.material-ui.com/#/)'s menu is better organized.
   - Hovewer [1.0.0-alpha.11's Documentation](https://material-ui-1dab0.firebaseapp.com/) has better rooting, which may not included with Material UI it self. Plus it's navigation also feels faster.

Thank you for your feedback in advance!

**Comment by oliviertassinari** [comment]
> Should the npm build command work on a local environment?

Yes.

> Do you have an example project similar to this one which supports server side rendering and maybe HTML and CSS minification too?

You can have a look at https://github.com/oliviertassinari/SplitMe

> Which version of Material-UI is the newest 0.17.4 or 1.0.0-alpha.11?

The newest is always the one with the high version number. But one is a prerelease version while the other one is stable.

**Comment by kireerik** [comment]
I see. It doesn't give any error or message, but the server doesn't runs at least not on port 80 or 3000.

Thank you! I will, but I would need something naked.

I see. So 1.0.0-alpha.11 (the current prerelease) is the newest which considered unstable, right?
- Are you planning to keep and or fix the features especially the chekboxes' animations I like about the current stable release in the long run (like in one of the next prereleases)?

**Comment by oliviertassinari** [comment]
> I see. It doesn't give any error or message, but the server doesn't runs at least not on port 80 or 3000.

The `npm run build` script doesn't start any server, it only outputs the files.

>  I would need something naked.

Please let us know if you find one, that could help others.

> Are you planning to keep and or fix the features especially the chekboxes' animations

We don't have any plan on the dimension. Might be something that needs to be improved.

**Comment by kireerik** [comment]
I see. The `npm run build` command do work, but the `npm build` command doesn't. I think this was my mistake or should it work without the `run` command as well?

All right, I think I will fork this repository soon and start to develop such one, because I will need one so I can start to work an a new project of mine after I have such a setup. Which will be a more current version of [Search is Back!](https://searchisback.com/) with more possibilities by the way.

Could you please clarify this? What do you mean by dimension? So these issues I listed previously are potential bugs and probably will be fixed over time?

Also I have found a Vue.js based library called [Muse-UI](http://www.muse-ui.org/?lang=en#/index) which is really similar to Material UI. For example their [checkbox](http://www.muse-ui.org/?lang=en#/checkbox) animation looks identical to your's. I didn't decided yet which one to choose. I will might use this other one because Vue.js' [performance](http://www.stefankrause.net/js-frameworks-benchmark5/webdriver-ts/table.html) is closer to Vanilla JS' performance.

**Comment by oliviertassinari** [comment]
>  I think this was my mistake or should it work without the run command as well?

That's an *npm* specificity only a few scripts works without `run`. That's no longer true with `yarn`.

@kireerik Thanks for sharing *muse-ui*, that's interesting! I would recommend you starting with [create-react-app](https://github.com/facebookincubator/create-react-app).
Performance is definitely not the only aspect to look at. Good luck picking one. Hard choice.

**Comment by kireerik** [comment]
I see.

You are welcome!

Thank you for recommending it. It's philosophy is great and I like that approach, but unfortunately it has lots of limitations. I need server side rendering and CSS Modules or something to handle style resources properly. Also after an `eject` the project set up looks too robust obviously. So I think it is better if I start with a fork.

I know, it is just the most important aspect for me. Yes, it is a really hard decision. I like Vue.js's concept more than React. However I think I will go with Material UI anyway, because it looks more finished than @myronliu347's Muse UI and it also has touch support for the Drawer component, even tough if it is lagging.

> We don't have any plan on the dimension. Might be something that needs to be improved.
> > Could you please clarify this? What do you mean by dimension? So these issues I listed previously are potential bugs and probably will be fixed over time?

Could you please answer my above questions as well? (A short yes or no is enough for me.)

Thank you for your feedback in advance!

**Comment by mbrookes** [comment]
@kireerik Muse-UI is a port of Material-UI to Vue, hence the similarities.

**Comment by kireerik** [comment]
Hi @mbrookes, thank you! I saw that Material-UI is older, but there was no reference to it in Muse UI.

Can you add anything about the [issues in the alpha release](https://github.com/callemall/material-ui-webpack-example/issues/9#issuecomment-296235046)?

**Comment by mbrookes** [comment]
> Which version of Material-UI is the newest 0.17.4 or 1.0.0-alpha.11?

Neither.  1.0.0-alpha is a work-in-progress migration, 0.17.4 is in maintenance mode, receiving (mostly) just fixes. Both are receiving regular releases.

> I know this may sounds a silly question, but with 1.0.0-alpha.11 the Checkboxes' animations become less impressive. I really like their animation in 0.17.4.

I'm inclined to agree, but can't comment on the reason. @nathanmarks might have a view.

> Also the left side bar menu (called Drawers) become not swipeable on touch enabled devices with 1.0.0-alpha.11.

The Drawer implementation is only a placeholder to get the docs site running, it hasn't been fully migrated.

> Plus I think 0.17.4's Documentation's menu is better organized.

They're practically the same other than moving the API docs to separate pages. What's mossing is the dropdown arrow. But again, it's all WIP.

**Comment by kireerik** [comment]
Thank you @mbrookes for your detailed explanations! It is good to know that alpha is just a work in progress version.

@oliviertassinari Regarding server side rendering I have found [Razzle](https://github.com/jaredpalmer/razzle) and I have built an example project upon it called [Razzle Material UI Styled Example](https://github.com/kireerik/razzle-material-ui-styled-example).

I have also submitted a [pull request](https://github.com/callemall/material-ui/pull/6862) to add my example to the other ones. I hope you like it and will accept the request.

It was really hard to find a React starter project which were at least a little close to my needs, I have tested and tried so many of them.
